### PR TITLE
Fix empty label radix bug

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Fix empty label radix bug
- Check validity of input labels

Related issues: 
https://github.com/ainblockchain/ain-blockchain/issues/552
https://github.com/ainblockchain/ain-blockchain/issues/519

### CODE REVIEW

File: db/radix-tree.js [hunk 1]
```diff
@@ -91,21 +91,42 @@ class RadixTree {
...
-        internalNode.setChild(newChildLabel.charAt(0), newChildLabel.slice(1), newChild);
-
-        return newChild;
+        if (commonPrefix.length === labelSuffix.length) {
+          // Insert an internal node between curNode and the existing child.
+          const internalNode = new RadixNode();
+          curNode.setChild(labelRadix, commonPrefix, internalNode);
+
+          // Insert the existing child node as a child of the internal node.
+          const childNewLabel = childLabelSuffix.slice(commonPrefix.length);
```
Comment (liayoo) [inline db/radix-tree.js:103]: It seems that this sequence of ops (getting a label, getting a label radix, getting a label suffix, inserting as a child) is repeated many times. Could you make a function for this?

File: db/radix-tree.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ class RadixTree {
     let curNode = this.root
     let labelIndex = 0;
     while (labelIndex < hexLabel.length) {
-      const labelRadix = hexLabel.charAt(labelIndex);
+      const labelRadix = hexLabel.slice(labelIndex, labelIndex + 1);
```
Comment (liayoo) [inline db/radix-tree.js:76]: question) is this change for a performance increase?

**Review by liayoo — APPROVED** [review]
LGTM with some comments & questions. Thanks!

File: db/radix-tree.js (hunk 1 above)
Comment (platfowner) [reply db/radix-tree.js:109]: Added _setChildWithLabel() util function. Thanks!

File: db/radix-tree.js (hunk 2 above)
Comment (platfowner) [reply db/radix-tree.js]: No, it's just a result of debugging trials. Just reverted. Thanks!

**Pull request merged**